containers/gds/Dockerfile

```diff
@@ -1,22 +1,51 @@
-FROM golang:1.16 AS builder
+# Dynamic Builds
+ARG BUILDER_IMAGE=golang:1.16-buster
+ARG FINAL_IMAGE=debian:buster-slim
 
-WORKDIR /srv/build
+# Build stage
+FROM ${BUILDER_IMAGE} as builder
 
-COPY . .
-RUN go test ./... && go build -v ./cmd/gds
-RUN go build -v ./cmd/sectigo
-RUN go build -v ./cmd/gdsutil
+# Ensure ca-certificates are up to date
+RUN update-ca-certificates
 
-FROM ubuntu:bionic
+WORKDIR $GOPATH/src/github.com/trisacrypto/directory
+
+# Use modules for dependencies
+COPY go.mod .
+COPY go.sum .
+
+ENV CGO_ENABLED=0
+ENV GO111MODULE=on
```

Comment:
Would you please clarify the usage of these new lines?

Reply:
`CGO_ENABLED=0` turns CGO compilation off, which means that Go will generate a statically compiled binary with no external dependencies (runtime linking to C libraries). This is a slightly slower compilation process and requires that we have only pure Go packages and dependencies that don't use C libraries; however, the binary can then be run completely independently in a very lightweight environment without having to install any other libraries (and reduces the overall size of the container). 

`GO111MODULE=on` forces Go to use `go module` even if the source code is in the `$GOPATH` (which it is, see line 11). This is another future proofing thing to make it easier to upgrade to Go 1.18 which will require go modules.